bundles/org.openhab.binding.coolmasternet/src/main/java/org/openhab/binding/coolmasternet/internal/handler/HVACHandler.java

```diff
@@ -162,8 +161,8 @@ public void handleCommand(final ChannelUID channelUID, final Command command) {
                             final QuantityType<?> value = new QuantityType<>(temp, controller.getUnit());
                             updateState(SET_TEMP, value);
                         }
-                    } else if (command instanceof DecimalType) {
-                        final DecimalType temp = (DecimalType) command;
+                    } else if (command instanceof QuantityType) {
+                        final QuantityType<?> temp = (QuantityType) command;
                         controller.sendCommand(String.format("temp %s %s", uid, temp));
```

Comment:
Don't you need to make sure to convert the quantity to the correct units before sending the command?

Reply:
Done. I also added 1 decimal place formatting, which makes it consistent with the `temp` command in the CoolMasterNet Programmer Reference Guide.

Tested fine:
```
2020-05-15 09:26:02.433 [TRACE] [masternet.internal.ControllerHandler] - Waiting 25 ms for buffer to sink
2020-05-15 09:26:02.433 [TRACE] [masternet.internal.ControllerHandler] - Buffer empty following unsolicited > prompt
2020-05-15 09:26:02.433 [TRACE] [masternet.internal.ControllerHandler] - Sending command 'temp L4.000 23.0'
2020-05-15 09:26:02.434 [TRACE] [masternet.internal.ControllerHandler] - Read result 'OK'
......
2020-05-15 09:26:11.272 [TRACE] [masternet.internal.ControllerHandler] - Waiting 25 ms for buffer to sink
2020-05-15 09:26:11.298 [TRACE] [masternet.internal.ControllerHandler] - Buffer empty following solicited > prompt
2020-05-15 09:26:11.298 [TRACE] [masternet.internal.ControllerHandler] - Sending command 'query L4.000 t'
2020-05-15 09:26:11.299 [TRACE] [masternet.internal.ControllerHandler] - Read result '23'
2020-05-15 09:26:11.300 [TRACE] [masternet.internal.ControllerHandler] - Read result 'OK'

```